src/olympia/discovery/utils.py

```diff
@@ -0,0 +1,45 @@
+import json
+
+from django.conf import settings
+
+import requests
+from django_statsd.clients import statsd
+
+import olympia.core.logger
+from olympia import amo
+from olympia.addons.models import Addon
+
+from . import data
+
+
+log = olympia.core.logger.getLogger('z.amo')
+
+
+def call_recommendation_server(telemetry_id, endpoint):
+    log.debug(u'Calling recommendation server: {0}'.format(endpoint))
+    with statsd.timer('services.recommendations'):
+        response = requests.post(
+            endpoint,
+            timeout=settings.SIGNING_SERVER_TIMEOUT)
+    if response.status_code != 200:
+        msg = u'Calling recommendation engine failed: {0}'.format(
+            response.reason)
+        log.error(msg)
+        return []
+    return json.loads(response.content).get('results', [])
+
+
+def get_recommendations(telemetry_id):
+    url = settings.RECOMMENDATION_ENGINE_URL + telemetry_id

```

Comment:
I'm confused why you build the `url` in `get_recommendations()` and still pass both the `telemetry_id` and the `url` to the `call_recommendation_server()` function. I'd say the `get_recommendations()` function does not need to know the `url`, and instead it should just call `call_recommendation_server(telemetry_id)` and have *that* function build the `url`.

Reply:
passing _both_ the `url` and `telemetry_id` was a mistake - it should just have been the url.  But sure, can just pass the id.

Reply:
https://github.com/mozilla/addons-server/pull/6324/commits/5d695e8ac12962936d1e3bfedbbe321ad5d2709f